File: backend/src/main/java/com/hardware/erp/branch/service/BranchService.java
package com.hardware.erp.branch.service;

import com.hardware.erp.branch.dto.BranchDtos.AssignUserBranchRequest;
import com.hardware.erp.branch.dto.BranchDtos.BranchRequest;
import com.hardware.erp.branch.dto.BranchDtos.BranchResponse;
import com.hardware.erp.branch.dto.BranchDtos.BranchStockResponse;
import com.hardware.erp.branch.dto.BranchDtos.BranchSummaryResponse;

import java.time.LocalDate;
import java.util.List;

/**
 * CR-092. Listing branches is free (every shop has its MAIN one and the
 * name is printed on documents); creating a second one, transfers and the
 * per-branch views are gated on FeatureKey.MULTI_BRANCH inside the service,
 * so an internal caller is refused exactly like an HTTP one.
 */
public interface BranchService {

    List<BranchResponse> list();

    BranchResponse get(Long id);

    BranchResponse create(BranchRequest request);

    BranchResponse update(Long id, BranchRequest request);

    List<BranchSummaryResponse> summary(LocalDate from, LocalDate to);

    List<BranchStockResponse> stock(Long branchId, String search);

    BranchResponse assignUser(Long userId, AssignUserBranchRequest request);
}

File: frontend/src/modules/branch/services/branchService.ts
import { apiGet, apiPost, apiPut } from '@/services/apiClient';
import type { PageResponse } from '@/shared/types/api';
import type {
  BranchRequest, BranchResponse, BranchStockResponse, BranchSummaryResponse,
  StockTransferRequest, StockTransferResponse,
} from '../types';

/** Backend: branch/controller/BranchController.java (CR-092) */
export const branchService = {
  list: () => apiGet<BranchResponse[]>('/v1/branches'),
  create: (body: BranchRequest) => apiPost<BranchResponse>('/v1/branches', body),
  update: (id: number, body: BranchRequest) => apiPut<BranchResponse>(`/v1/branches/${id}`, body),
  summary: (from: string, to: string) =>
    apiGet<BranchSummaryResponse[]>('/v1/branches/summary', { params: { from, to } }),
  stock: (branchId?: number | null, search?: string) =>
    apiGet<BranchStockResponse[]>('/v1/branches/stock', { params: { branchId: branchId ?? undefined, search: search || undefined } }),
  transfers: (page = 0, size = 20) =>
    apiGet<PageResponse<StockTransferResponse>>('/v1/branches/transfers', { params: { page, size } }),
  transfer: (body: StockTransferRequest) => apiPost<StockTransferResponse>('/v1/branches/transfers', body),
  assignUser: (userId: number, branchId: number | null) =>
    apiPut<BranchResponse | null>(`/v1/branches/users/${userId}`, { branchId }),
};
